presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java

```diff
@@ -15,259 +15,195 @@
-import static com.facebook.airlift.http.client.Request.Builder.prepareGet;
-import static com.facebook.airlift.http.client.Request.Builder.preparePost;
-import static com.facebook.airlift.http.client.StaticBodyGenerator.createStaticBodyGenerator;
-import static com.facebook.presto.execution.TaskInfo.createInitialTask;
 import static com.facebook.presto.execution.buffer.OutputBuffers.BufferType.PARTITIONED;
 import static com.facebook.presto.execution.buffer.OutputBuffers.createInitialEmptyOutputBuffers;
-import static com.facebook.presto.server.RequestHelpers.setContentTypeHeaders;
-import static com.facebook.presto.server.smile.AdaptingJsonResponseHandler.createAdaptingJsonResponseHandler;
-import static com.facebook.presto.spi.StandardErrorCode.NATIVE_EXECUTION_TASK_ERROR;
-import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
-import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
 /**
- * NativeExecutionTask provide the abstraction of executing tasks via c++ worker.
- * The plan is to use it for native execution of Presto on Spark.
- * The plan and splits is provided at creation of NativeExecutionTask, it doesn't
- * support adding more splits during execution.
+ * NativeExecutionTask provide the abstraction of executing tasks via c++ worker. It is used for native execution of Presto on Spark. The plan and splits is provided at creation
+ * of NativeExecutionTask, it doesn't support adding more splits during execution.
+ * <p>
+ * Caller should manage the lifecycle of the task by exposed APIs. The general workflow will look like:
+ * 1. Caller shall start() to start the task. The task finish signal can be told by the returned future.
+ * 2. Caller shall call getTaskInfo() any time to get the current task info. Until the caller calls stop(), the task info fetcher will not stop fetching task info.
+ * 3. Caller shall call pollResult() continously to poll result page from the internal buffer. The result fetcher will stop fetching more results if buffer hits its memory cap,
+ * until pages are fetched by caller to reduce the buffer under its memory cap.
+ * 4. Caller must call stop() to release resource.
  */
 public class NativeExecutionTask
 {
     private static final Logger log = Logger.get(NativeExecutionTask.class);
-    private static final int GET_TASK_INFO_INTERVALS = 100;
-    private static final String LOCAL_URI = "http://127.0.0.1/v1/task/";
 
     private final Session session;
     private final TaskId taskId;
     private final PlanFragment planFragment;
     private final OutputBuffers outputBuffers;
-    private final HttpClient httpClient;
+    private final PrestoSparkHttpWorkerClient workerClient;
     private final TableWriteInfo tableWriteInfo;
-    private final RemoteTaskStats stats;
     private final List<TaskSource> sources;
     private final Executor executor;
     private final ScheduledExecutorService updateScheduledExecutor;
```

Comment:
Can we convert it into a local var? Check comment: https://github.com/prestodb/presto/pull/18458/files#r989703377

Reply:
Missed it